### ISSUE TITLE
`hyperref` makes `varioref`'s `\vrefformat` no-op

### ISSUE THREAD

**Issue by gusbrs** [latex3/hyperref#225]
Currently, `hyperref` makes `varioref`'s user customization option `\vrefformat` no-op, as shown in the MWE below:

```latex
\documentclass{article}

\usepackage[nospace]{varioref}
\usepackage{hyperref}

\renewcommand*{\vrefformat}[2]{%
  Foo \ref{#2} \vpageref[#1]{#2}}

\begin{document}

\section{Section 1}
\label{sec:section-1}

\clearpage{}

\vref{sec:section-1}

\end{document}
```
Renders:

![Screenshot from 2022-01-31 17-06-08](https://user-images.githubusercontent.com/19410606/151864888-d5ec1898-fabe-4d1f-b081-23c2b05da4ea.png)

Commenting `hyperref` out of the above document renders:

![Screenshot from 2022-01-31 17-07-41](https://user-images.githubusercontent.com/19410606/151865041-0bf07b12-b067-4683-bfa0-8334e9c525d0.png)

This happens because `hyperref` redefines `\vr@f` and `\vref@star` (I haven't tested, but I presume the same problem occurs with `\Vref` for the same reason). And does so without using `\vrefformat` (and `\Vrefformat`), but hard-coding the formats in the redefinitions it makes.

As far as I can tell, this is done at: https://github.com/latex3/hyperref/blob/c25f44c5d880a5897376d947957e7a129b376e6c/hyperref.dtx#L13027-L13058

I don't know if there's a reason for this that I've missed, but it seems possible, at first glance, to use `\vrefformat` instead of hard-coding it (maybe redefining `\vrefdefaultformat` for this).  And thus not override this user option from `varioref`.

**Comment by u-fischer** [comment]
yes it is wrong, but looking at it I think it would be better if the support for links would be done by varioref directly and hyperref would stop to patch (and break)  it.  @FrankMittelbach ?

**Comment by FrankMittelbach** [comment]
yes the right place is in varioref to offer support for hyperref not the other way around, but that requires sync of the two (which was more or less impossible in the past with hyperref just patching all over the place. :-)). We can do this now, now that we control both sides ... you have write acces to varioref too  ...  we should put that on the agenda for a change in the not too distant future.

**Comment by u-fischer** [comment]
I can easily directly surround all the hyperref (and nameref) code with some test that disables it under some condition that varioref can then activate, `\ifdefined\vp@nohyperrefpatches`. The only thing I find a bit bewildering is why hyperref (re)defines `\Ref`  only if varioref is loaded, this looks rather odd, I guess that should be moved outside first.

**Comment by FrankMittelbach** [comment]
> The only thing I find a bit bewildering is why hyperref (re)defines `\Ref` only if varioref is loaded, this looks rather odd, I guess that should be moved outside first.

historical? if i remember correctly `\Ref` wasn't in the kernel initially
